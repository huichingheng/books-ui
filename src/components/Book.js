import React, { Component } from "react";

const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";

class Book extends Component {
  constructor() {
    super();
    this.state = {
      books: []
    };
  }

  async getBook() {
    const url = `${API_HOST}/books`;
    const response = await fetch(url);
    const data = await response.json();
    this.setState({
      books: data
    });
  }

  async componentDidMount() {
    await this.getBook();
  }

  render() {
    return (
      <div>
        <h2>Books</h2>
        {this.state.books.map(book => {
          return <li key={book._id}>{book.title}</li>;
        })};
      </div>
    );
  }
}

export default Book;
