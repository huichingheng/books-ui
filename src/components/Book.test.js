import React from "react";
import ReactDOM from "react-dom";
import Book from "./Book";
import ShallowRender from "react-test-renderer/shallow";

test("renders Book component", () => {
  const renderer = new ShallowRender();
  renderer.render(<Book />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
