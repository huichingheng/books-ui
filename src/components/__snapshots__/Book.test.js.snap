// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Book component 1`] = `
<div>
  <h2>
    Books
  </h2>
  ;
</div>
`;
